### ISSUE TITLE
Configuration with nested objects

### ISSUE THREAD

**Issue by WixoLeo** [chaostoolkit/chaostoolkit#198]
**Fantastic day everyone!**

Currently it's not possible to have a configuration with nested objects inside `experiment.json`.
Example:

```json
"configuration": {
    "servers": {
      "north": {
        "ip": "123.123.123.123",
        "port": "1234",
        "id": "f43e345345",
        "dashboard_id": "234234324",
        ...
      },
      "east": {
        "ip": "123.123.123.124",
        "port": "1235",
        "id": "f43e345346",
        "dashboard_id": "234234327",
        ...
      },
      ...
    }
  },
```

Now imagine that each server has many properties, and perhaps some properties are objects as well.

So in order to solve this with the current version, we have to do one of these solutions:

1. In load the separate config in the controls (perhaps configure control) and somehow save it as a global object so everything would be able to access it.
2. Have each property be named like this `"servers_east_ip"`, `"servers_east_port"` which is really not an option because these objects can go deep.

Another thing to note, why we wouldn't want to use the 1st solution is because we want to inject this configuration when we run the experiment.

In our CLI script we have an option to run a sandbox or prod environment. Each one has it's own configuration file that we pass to the `chaos run` command using the `--var-file`.
Therefore it would be much easier if it worked with the deep objects out of the box.

What do you think?

**Comment by alexander-gorelik** [comment]
There is another option to put the huge servers object directly into prob/action argument but then, experiment JSON becoming huge and nasty!

**Comment by Lawouach** [comment]
IIRC, while not officially documented, the ctk is not unhappy if you set an object that way for config values. Have you tried just to see?

**Comment by WixoLeo** [comment]
> IIRC, while not officially documented, the ctk is not unhappy if you set an object that way for config values. Have you tried just to see?

Well the problem is that when we pass it in the arguments, it passes it as a string instead of dict

```
         "arguments": {
            "servers": "${servers}"
          }
```

How can we specify that we want to pass it as an object without parsing it in our code?

**Comment by Lawouach** [comment]
Only when using a var file right?

**Comment by WixoLeo** [comment]
> Only when using a var file right?

No, even if I don't use a var file, it's straight up in the configuration section (just to see where the root problem is)

**Comment by WixoLeo** [comment]
Here is a full example of a dummy experiment:
experiment.json:
```json
{
  "version": "1.0.0",
  "title": "test",
  "description": "test",
  "configuration": {
    "patho": "",
    "nested": {
      "onea": "fdsfdsf",
      "lol": {
        "haha": [
          1,
          2,
          3
        ]
      }
    }
  },
  "tags": [
    "network",
    "lala"
  ],
  "steady-state-hypothesis": {
    "title": "Check something",
    "probes": [
      {
        "name": "My probe",
        "description": "My best probe",
        "type": "probe",
        "tolerance": false,
        "provider": {
          "type": "python",
          "module": "playground.probes",
          "func": "test_nested",
          "arguments": {
            "nested": "${nested}"
          }
        }
      }
    ]
  },
  "method": [],
  "rollbacks": []
}
```

Probe:

```py
__all__ = [
    "test_nested",
]


def test_nested(nested: dict = None):
    print(nested["onea"]) # TypeError: string indices must be integers
    return False


```

I checked using the debugger, the "nested" is in fact a string

**Comment by Lawouach** [comment]
Example really appreciated! Thanks.

I'll look at it today.
